package.json

```diff
@@ -99,7 +99,7 @@
     "rollup-plugin-terser": "^7.0.2",
     "shelljs": "^0.8.2",
     "sinon": "4.5.0",
-    "typescript": "^3.7.0",
+    "typescript": "^4.2.3",
```

Comment:
adding documentation and increasing this version is probably not a good way to create a PR, because who knows what side effects this could have unrelated to the docs

Reply:
Yeah I know. 😔 I was having a hard time thinking about this because the new tests that I wrote literally won't work without the new TypeScript version -- particularly the failing tests.

I could remove the version update, but I'd also have to remove the new tests. I can separate them if desirable though.

Reply:
How does one update typescipt dependency? Does it have a consequence for the user?

Reply:
If it's truly a dev dependency and docs are new, maybe we just need to hint the version of typescript they depend on

Reply:
If I understand correctly, because TypeScript is a devDependency, updating our own TypeScript version shouldn't affect other users.

When the end user checks their types through something like `tsc`, _their_ version of TypeScript will be used.

So the potential for breaking changes would only be a concern if we changed type definitions. However, we haven't changed any type definitions -- only tests. And the `check:ts` script worked fine without having to modify any definitions.

Reply:
should be fine then, do people mark docs and examples with TS version required for an example to work? Maybe a good idea to add a label in such cases

Reply:
> If it's truly a dev dependency and docs are new, maybe we just need to hint the version of typescript they depend on

Maybe we could give a small reminder to use the latest version of TypeScript?

Thankfully, it's very unlikely that the user would feel any impact between v3 and v4.

Reply:
> should be fine then, do people mark docs and examples with TS version required for an example to work? Maybe a good idea to add a label in such cases


Interestingly enough, I don't see people mention TypeScript versions in docs very often -- if ever. This is even the case for popular libraries. There seems to be a general assumption that people are using the latest version of TS.

That might be because it's pretty uncommon to update TypeScript's major version and run into any errors -- especially any significant errors. Managing TS dependencies tends to be fairly easy.

Reply:
Alright, lets wait for someone saying example doesn't work, until then we avoid adding versions :)